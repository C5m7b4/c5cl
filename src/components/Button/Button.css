.c5-normal-button {
  box-shadow: 2px 4px 6px rgba(0, 0, 0, 0.3);
}

.c5-normal-button:hover {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.3);
}

.c5-normal-button:active {
  transform: scale(0.96);
}

.c5-normal-button-small {
  border-radius: 10px;
  padding: 8px 15px;
  cursor: pointer;
  border: 0;
  outline: none;
}

.c5-normal-button-medium {
  border-radius: 10px;
  padding: 12px 20px;
  cursor: pointer;
  border: 0;
  outline: none;
  font-size: 1.3rem;
}

.c5-normal-button-large {
  border-radius: 10px;
  padding: 20px 40px;
  cursor: pointer;
  border: 0;
  outline: none;
  font-size: 2rem;
}
