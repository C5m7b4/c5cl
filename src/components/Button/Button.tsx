import React, { useState, useEffect } from 'react';
import './Button.css';

export type NormalButtonType =
  | 'normal'
  | 'green'
  | 'purple'
  | 'danger'
  | 'info'
  | 'success'
  | 'error';

export type NormalButtonSize = 'small' | 'medium' | 'large';

export interface ButtonProps {
  label: string;
  style?: React.CSSProperties;
  onClick?: () => React.MouseEvent<HTMLButtonElement>;
  disabled?: boolean;
  type?: NormalButtonType;
  size?: NormalButtonSize;
}

const Button = (props: ButtonProps) => {
  const [hover, setHover] = useState(false);

  const {
    label,
    style,
    onClick,
    disabled = false,
    type,
    size = 'small',
  } = props;
  let buttonStyle = {};

  switch (type) {
    case 'normal':
      buttonStyle = {
        backgroundColor: hover ? '#877b87' : '',
        color: hover ? '#fff' : '#000',
      };
      break;
    case 'green':
      buttonStyle = {
        backgroundColor: hover ? '#2ea62e' : 'limegreen',
        color: '#fff',
      };
      break;
    case 'purple':
      buttonStyle = {
        backgroundColor: hover ? '#9c229c' : '#ba1eba',
        color: '#fff',
      };
      break;
    case 'danger':
      buttonStyle = {
        backgroundColor: hover ? '#c9a734' : '#e8bb25',
        color: '#fff',
      };
      break;
    case 'info':
      buttonStyle = {
        backgroundColor: hover ? '#5680a8' : '#5a95cd',
        color: '#fff',
      };
      break;
    case 'success':
      buttonStyle = {
        backgroundColor: hover ? 'rgb(35 169 71)' : '#1bcc4a',
        color: '#fff',
      };
      break;
    case 'error':
      buttonStyle = {
        backgroundColor: hover ? '#bf2424' : 'red',
        color: '#fff',
      };
      break;
    default:
    // do nothing;
  }

  buttonStyle = {
    ...buttonStyle,
    transition: 'background-color .3s ease, color .3s ease',
  };

  const toggleHover = () => {
    setHover(!hover);
  };

  return (
    <button
      onClick={onClick}
      style={{ ...style, ...buttonStyle }}
      disabled={disabled}
      onMouseEnter={toggleHover}
      onMouseLeave={toggleHover}
      className={`c5-normal-button c5-normal-button-${size}`}
    >
      {label}
    </button>
  );
};

export default Button;
