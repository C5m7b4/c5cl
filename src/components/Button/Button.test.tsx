import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Button from './Button';

describe('Button', () => {
  test('renders the button component', () => {
    render(<Button label="Click Me" />);
  });
  test('should render without crashing', () => {
    const frag = render(<Button label="clickme" />);
    expect(frag).toMatchSnapshot();
  });
  test('should handle normal type', () => {
    const { container } = render(<Button label="clickme" type="normal" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle green type', () => {
    const { container } = render(<Button label="clickme" type="green" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle purple type', () => {
    const { container } = render(<Button label="clickme" type="purple" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle danger type', () => {
    const { container } = render(<Button label="clickme" type="danger" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle info type', () => {
    const { container } = render(<Button label="clickme" type="info" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle success type', () => {
    const { container } = render(<Button label="clickme" type="success" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle error type', () => {
    const { container } = render(<Button label="clickme" type="error" />);
    const btn = container.querySelector('button') as HTMLButtonElement;
    fireEvent.mouseEnter(btn);
    expect(container).toMatchSnapshot();
  });
});
