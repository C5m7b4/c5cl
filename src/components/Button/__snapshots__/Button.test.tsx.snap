// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should handle danger type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="background-color: rgb(201, 167, 52); color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease;"
  >
    clickme
  </button>
</div>
`;

exports[`Button should handle error type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="background-color: rgb(191, 36, 36); color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease;"
  >
    clickme
  </button>
</div>
`;

exports[`Button should handle green type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="background-color: rgb(46, 166, 46); color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease;"
  >
    clickme
  </button>
</div>
`;

exports[`Button should handle info type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="background-color: rgb(86, 128, 168); color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease;"
  >
    clickme
  </button>
</div>
`;

exports[`Button should handle normal type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease; background-color: rgb(135, 123, 135);"
  >
    clickme
  </button>
</div>
`;

exports[`Button should handle purple type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="background-color: rgb(156, 34, 156); color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease;"
  >
    clickme
  </button>
</div>
`;

exports[`Button should handle success type 1`] = `
<div>
  <button
    class="c5-normal-button c5-normal-button-small"
    style="background-color: rgb(27, 204, 74); color: rgb(255, 255, 255); transition: background-color .3s ease, color .3s ease;"
  >
    clickme
  </button>
</div>
`;

exports[`Button should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="c5-normal-button c5-normal-button-small"
        style="transition: background-color .3s ease, color .3s ease;"
      >
        clickme
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="c5-normal-button c5-normal-button-small"
      style="transition: background-color .3s ease, color .3s ease;"
    >
      clickme
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
